merkledag/ipld.md

```diff
@@ -308,6 +308,116 @@ In order to preserve merkle-linking's power, we must ensure that there is a sing
 
 **The IPLD Canonical format is _canonicalized CBOR_.**
 
+## Relationship with Protocol Buffers legacy IPFS node format
+
+IPLD has a known conversion with the legacy Protocol Buffers format. This format is defined with the Protocol Buffers syntax as:
+
+    message PBLink {
+        optional bytes  Hash = 1;
+        optional string Name = 2;
+        optional uint64 Tsize = 3;
+    }
+    
+    message PBNode {
+        repeated PBLink Links = 2;
+        optional bytes  Data = 1;
+    }
+
+The conversion to the IPLD data model must have the following properties:
+
+- It should be convertible back to protocol buffers, resulting in an identical byte stream (so the hash corresponds). This implies that ordering and duplicate links must be preserved in some way.
+- When using paths as defined earlier in this document, links should be accessible without further indirection. This requires the top node object to have keys corresponding to link names.
+- Link names should not conflict with other keys.
```

Comment:
maybe use caps for "SHOULD NOT", but note that many objects already do. so we define an access order: "in case of a conflict, use the first one to appear in the protobuf."